File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Strikethrough.kt
package com.nouprax.markdown.core

public class Strikethrough internal constructor(
    public val content: kotlin.collections.List<Markup>,
    override val scope: Scope,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visitStrikethrough(this)
}

File: packages/swift-markdown-core/Sources/MarkdownCore/Markup/Strikethrough.swift
import MarkdownCoreC

/// Struck-through text. Requires the `strikethrough` extension.
public struct Strikethrough: Markup {
    /// Where it is. See ``Scope`` — boundaries, not a byte range.
    public let scope: Scope
    /// The struck-through inline content.
    public let content: [any Markup]

    /// Dispatches to the visitor's `Strikethrough` case.
    public func accept<V: MarkupVisitor>(_ visitor: inout V) -> V.Result { visitor.visit(self) }
}

extension Strikethrough {
    init(from node: OpaquePointer, content: [any Markup]) {
        self.init(scope: Self.scope(from: node), content: content)
    }
}
